skipoffset added to AttachPodAds

JinHedman opened the pull request:
We added an attribute to the attachLinear method in AttachPodAds. We set it as 5 seconds but it could be changed to another value or a percentage. Right now it's very straightforward, every ad will be skippable after 5 seconds, assuming they're longer than 5 seconds. We could also make it so that we look for a skipoffset value from the ads properties and if they have one we'll add it to the params of the builder so you can choose skipoffset value or not have one at all. 

Also a small sidenote, the skipoffset attribute for the <Linear> tag is not part of VAST 2.0. The builder will still create a VAST 2.0 XML to the official documentation, that is to say it wont add the attribute to the <Linear> tag if you choose VAST 2.0.

Nfrederiksen reviewed (changes requested):
Am I missing something here? (: 
I only see one line changed. Are there any other commits to be check in? 
The PR description makes it sound like you've provided a way to set the skipoffset or?

JinHedman commented:
Hello my apologies, I did not understand that the skipoffset was supposed to be a parameter that the user could input a value to. I've changed it now! I added a isValidSkipOffset in the file vast-maker.js but I'm not sure that it's the best place to put it in! Just holla at me if I should move it elsewhere!

JinHedman commented:
Thank you for the feedback @Nfrederiksen. I'll try and have these changes implemented by tomorrow! You've been a huge help for the student development team this iteration, thank you so much! 🙏🏼

JinHedman commented:
Heya, I added the skipoffset to the vmap-maker by storing the skipoffset in the params that are sent in to the VastBuilder. I also updated it so that a parameter for skipoffset exists in the vmap endpoint. I also changed the name of isValidSkipOffsetValue to getSkipOffsetValue and added so that you can write the skipoffsets in seconds without having to write it in the "hh:mm:ss" format!

Nfrederiksen commented on utils/vmap-maker.js:
```diff
@@ -90,6 +91,7 @@ function VmapBuilder(params) {
 
   const breakpoints = params.breakpoints ? params.breakpoints.split(",").filter((item) => !isNaN(Number(item))) : [];
   if (params.preroll) {
+    //console.log("PARAMS: ",params.generalVastConfigs);
```
This may be removed/cleaned up

Nfrederiksen commented on utils/vast-maker.js:
```diff
@@ -351,6 +352,30 @@ function indexOfSmallest(a) {
   return lowest;
 }
 
+// Validate that params.skipoffset is a valid VAST skipoffset value ("x%" or "hh:mm:ss").
```
You can remove this comment too

Nfrederiksen reviewed (commented):
In Summary, Other than some need for clean up and timecode format disabling. The PR looks good.

Nfrederiksen reviewed (approved):
Look Good To Me! 
Sorry that the final review took some time.
